src/test/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpStreamTest.java

```diff
@@ -205,6 +206,26 @@ public void bigDataWithInvalidChecksum_throwsIOExceptionAfterCreateOnEof() throw
     }
   }
 
+  @Test
+  public void bigDataTruncated_throwsExpectedError() throws Exception {
+    byte[] bigData = new byte[HttpStream.PRECHECK_BYTES + 70001];
+    randoCalrissian.nextBytes(bigData);
+    byte[] truncated = Arrays.copyOfRange(bigData, 0, bigData.length - 1);
```

Comment:
A nice to have test coverage can be added.  A test case or test data for `bigData.length + 1` to cover the other half of the not equal check at https://github.com/bazelbuild/bazel/pull/13918/files#diff-e49ed82007084b0b163fc77f9f8a46b5336587b9ea739ad5bfb01a4f7eed8befR57.

Reply:
Added. Thanks!